Automattic/jetpack
# Boost: Insecure ajax api and verify current user capabilities

Fixes #21136

#### Changes proposed in this Pull Request:

* Changed nonce `AJAX_NONCE` to be action specific thus renamed to `DISMISS_CSS_RECOMMENDATION_NONCE`
* Checking current user capabilites before dismiss advanced css recommendation.
* Using `wp_send_json_error` instead of `wp_die`

#### Jetpack product discussion NA



#### Does this pull request change what data or activity we track or use? NA



#### Testing instructions:





As mentioned in the [comment](https://github.com/Automattic/jetpack/issues/21136#issuecomment-923704692):
1. You'd need Boost to give you an Advanced Critical CSS Recommendation. That occurs when it fails to generate Critical CSS for an area of your site, but not all of your site. The easiest way to do that is with a plugin that causes all posts with a custom post_type to redirect to an external URL - such as [simple-urls](https://wordpress.org/plugins/simple-urls/).
2. Once you have an advanced recommendation, try using the X button to dismiss it. That should trigger one of the AJAX calls. Once all Advanced recommendations have been dismissed, a link will appear which lets you trigger the other AJAX call; resetting the dismissals.

cc @thingalon
